dynamic/msgregistry/message_registry.go

```diff
@@ -22,6 +22,42 @@ import (
 
 const googleApisDomain = "type.googleapis.com"
 
+// ErrUnexpectedEnum is returned if the URL that was requested
+// resolved to an enum instead of a message.
+type ErrUnexpectedEnum struct {
+	URL                string
+	FullyQualifiedName string
+}
+
+func (e *ErrUnexpectedEnum) Error() string {
+	return fmt.Sprintf("type for URL %q is the wrong type: wanted message, got enum", e.URL)
+}
+
+// Is determines how ErrUnexpectedEnum is compared
+// against other errors when using errors.Is.
+func (e *ErrUnexpectedEnum) Is(err error) bool {
```

Comment:
From my read of the doc for `errors.Is` and what this optional method is intended to support, this doesn't feel right to me.

Instead of treating all such `ErrUnexpectedEnum` errors equivalent like this, I think we should instead omit this method and then callers can use `errors.As` instead:

```go
if unexpectedType := (*ErrUnexpectedEnum)(nil); errors.As(err, &unexpectedType) {

```

Reply:
Let me try that!